### DISCUSSION TITLE
Rueidis with Valkey

### DISCUSSION THREAD

**Discussion by Harleen13Singh** [redis/rueidis#685] [Q&A] [answered]
Hi folks, I was considering using [Valkey](https://github.com/valkey-io/valkey) instead of redis from AWS since it is cheaper and can save cost. I don't have much in depth knowledge of rueidis, but I don't want to use go-redis since I also has faced the connection pooling issue. Just wanted to clarify the functionality of library will be similar if I use Valkey?

**Comment by rueian** [comment, marked as answer]
Yes, this library doesn’t treat valkey specially. BTW what kind of connection issue you faced with go-redis?

**Reply by Harleen13Singh** [reply]
Thanks for the quick reply. Not exactly the connection issue. IIRC, it was related to when the context got cancelled and go-redis was not able to put back the existing connections back to the pool, and hence new connections were created, increasing the latency. I might be wrong here with the exact root cause, it was being discussed [here](https://github.com/redis/go-redis/issues/2046).

**Reply by rueian** [reply]
Thank you for telling me about that. I will look into the discussion. But I also need to say that this behavior can also happen on rueidis if the client doesn’t operate in pipeline mode, For example setting DisablePipeling to true.
